### PR TITLE
Implement Explanations in RPC

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?
We add Explanations, and the explain() query endpoint to the protocol. This enables all clients to retrieve all available explanations for a particular explainable concept map.

## What are the changes implemented in this PR?
* Implement `Explainable` and `Explainables` as part of `Answer` protocol
* Implement `Explanation` as part of `Logic` protocol
* Add `Explain.Req` and `Explain.Res` as part of the explain QueryManager endpoing

### CODE REVIEW

File: common/logic.proto [hunk 1]
```diff
@@ -106,3 +109,14 @@ message Rule {
         message Res {}
     }
 }
+
+message Explanation {
+    Rule rule = 1;
+    map<string, VarsList> varMapping = 2; // TODO perhaps this should be a "context" and contain more information about source
```
Comment (alexjpwalker) [inline common/logic.proto:115]: We need to use `snake_case` consistently everywhere for protocol field names - Java handles mixed casing relatively gracefully, but Python and JS don't.
```suggestion
    map<string, VarsList> var_mapping = 2; // TODO perhaps this should be a "context" and contain more information about source
```
(Other examples elsewhere in the PR)

**Review by alexjpwalker — APPROVED** [review]

File: common/answer.proto [hunk 2]
```diff
@@ -26,7 +26,24 @@ package grakn.protocol;
...
+
+message ExplainableOwnership {
+    string owner = 1;
+    string attribute = 2;
+    Explainable explainable = 3;
+}
+
+message Explainable {
+    string conjunction = 1;
+    int64 id = 2;
```
[the remark this answers is not in the document]
Comment (haikalpribadi) [reply common/answer.proto:46]: let's try `map<string, map<string, Explainable>>` if we can do it in GRPC

File: common/logic.proto [hunk 3]
```diff
@@ -106,3 +109,14 @@ message Rule {
         message Res {}
     }
 }
+
+message Explanation {
+    Rule rule = 1;
+    map<string, VarsList> var_mapping = 2; // TODO perhaps this should be a "context" and contain more information about source
+    ConceptMap then_answer = 3;
+    ConceptMap when_answer = 4;
```
Comment (haikalpribadi) [inline common/logic.proto:117]: `conclusion` and `condition` are better names

File: common/logic.proto [hunk 4]
```diff
@@ -106,3 +109,14 @@ message Rule {
...
+
+    message VarsList {
```
Comment (haikalpribadi) [inline common/logic.proto:119]: This should be `VarList`

**Review by haikalpribadi — CHANGES_REQUESTED** [review]

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**